src/csharp/Microsoft.Spark/Sql/RowConstructor.cs

```diff
@@ -119,16 +56,8 @@ internal void Reset()
             s_schemaCache?.Clear();
         }
 
-        /// <summary>
-        /// Get or cache the schema string contained in args. Calling this
-        /// is only valid if the child args contain the row values.
-        /// </summary>
-        /// <returns></returns>
-        private StructType GetSchema()
+        private static StructType GetSchema(string schemaString)
         {
-            Debug.Assert(s_schemaCache != null);
```

Comment:
We can still have this check (this tells you the prerequisite clearly)

Reply:
sure.  readded.